internal/config/validate.go

```diff
@@ -6,52 +6,57 @@ import (
 
 // validate checks if the Config struct is valid and returns InvalidConfigError if it's invalid.
 func (c *Config) validate() error {
-	var errs InvalidConfigError
 
 	if c.MaxRetries < 0 {
-		errs.appendFieldError("BUILDKITE_SPLITTER_RETRY_COUNT", "was %d, must be greater than or equal to 0", c.MaxRetries)
+		c.errs.appendFieldError("BUILDKITE_SPLITTER_RETRY_COUNT", "was %d, must be greater than or equal to 0", c.MaxRetries)
 	}
 
-	if got, min := c.Parallelism, 1; got < min {
-		errs.appendFieldError("BUILDKITE_PARALLEL_JOB_COUNT", "was %d, must be greater than or equal to %d", got, min)
-	}
+	if c.errs["BUILDKITE_PARALLEL_JOB"] == nil {
+		if got, min := c.NodeIndex, 0; got < 0 {
```

Comment:
Can we add a comment explaining why we are doing this for `PARALLEL_JOB` and `PARALLEL_JOB_COUNT`, but not for other envs? Also, I think the order of the validation matters here, maybe write that in the comment as well?

Reply:
That's good call. I added some comments. Let me know if it's still unclear 😀